### PR TITLE
Add generic web quick start

### PR DESCRIPTION
**Pull request by gewenyu99**
## What does this PR do?

Add a generic web quickstart. 
`/docs/quick-starts/js/`

Covers JS basics + Typescript basics.

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work.)

## Related PRs and Issues

(If this PR is related to any other PR or resolves any issue or related to any issue link all related PR and issues here.)

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

(Write your answer here.)

### CODE REVIEW

File: src/routes/docs/quick-starts/js/+page.markdoc [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+{% /only_dark %}
+{% only_light %}
+![Add a platform](/images/docs/quick-starts/add-platform.png)
+{% /only_light %}
+
+You can skip optional steps.
+
+{% /section %}
+{% section #step-2 step=2 title="Create project" %}
+You can install the Appwrite Web SDK using CDN with a script tag.
```
Comment (TGlide) [inline src/routes/docs/quick-starts/js/+page.markdoc:37]: That's not installing, right? Maybe `add` instead

File: src/routes/docs/quick-starts/js/+page.markdoc [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+```
+
+You can also install the Appwrite Web SDK using a package manager.
+```sh
+npm install appwrite
+```
+{% /section %}
+{% section #step-3 step=3 title="Import Appwrite" %}
+```js
+import { Client, Account} from 'appwrite';
```
Comment (TGlide) [inline src/routes/docs/quick-starts/js/+page.markdoc:50]:
```suggestion
import { Client, Account } from 'appwrite';
```

File: src/routes/docs/quick-starts/js/+page.markdoc [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+const authUser: Models.Session = await account.createEmailSession(email, password);
+```
+{% /section %}
+
+{% section #step-4 step=4 title="Extending TypeScript models" %}
+Sometimes you'll need to extend TypeScript models with your own type definitions.
+
+For example, when you fetch a list of documents from a collection, you can define the expected structure of the documents like this.
+```ts
+interface Idea extends Models.Document{
```
Comment (TGlide) [inline src/routes/docs/quick-starts/js/+page.markdoc:92]:
```suggestion
interface Idea extends Models.Document {
```

File: src/routes/docs/quick-starts/js/+page.markdoc [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+
+When you fetch documents, you can use this new `Idea` interface like this.
+
+```ts
+const response = await database.listDocuments(
+    ideasDatabaseId,
+    ideasCollectionId,
+    [Query.orderDesc("$createdAt"), Query.limit(queryLimit)]
+);
+const ideas: Idea[] = response.documents as Idea[];
```
Comment (TGlide) [inline src/routes/docs/quick-starts/js/+page.markdoc:107]:
```suggestion
const ideas = response.documents as Idea[];
```

**Review by TGlide — APPROVED** [review]

File: src/routes/docs/quick-starts/js/+page.markdoc (hunk 1 above)
Comment (divanov11) [reply src/routes/docs/quick-starts/js/+page.markdoc]: Agree with Thomas. Maybe say "One way to use the Appwrite Web SDK is with a CDN link". Or something along those lines.

Comment (gewenyu99) [reply src/routes/docs/quick-starts/js/+page.markdoc]: Changed to add

File: src/routes/docs/quick-starts/web/+page.markdoc [hunk 5]
```diff
@@ -0,0 +1,151 @@
+---
+layout: article
+title: Start with Web
+description: Build JavaScript or Typescript web apps with Appwrite and learn how to use our powerful backend to add authentication, user management, file storage, and more.
+difficulty: beginner
+readtime: 3
+back: /docs/quick-starts
+---
+
+Learn how to add Appwrite to your JavaScript or TypeScript web apps.
```
Comment (divanov11) [inline src/routes/docs/quick-starts/web/+page.markdoc:10]: This this is for the web I think javascript and typescript support can be assumed. How about: "Learn how to add Appwrite to your web apps."

File: src/routes/docs/quick-starts/web/+page.markdoc [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+{% /only_dark %}
+{% only_light %}
+![Add a platform](/images/docs/quick-starts/add-platform.png)
+{% /only_light %}
+
+You can skip optional steps.
+
+{% /section %}
+{% section #step-2 step=2 title="Create project" %}
+You can add the Appwrite Web SDK using CDN with a script tag.
```
Comment (divanov11) [inline src/routes/docs/quick-starts/web/+page.markdoc:37]: Shouldn't we encourage people to download the web SDK? 

I feel like the CDN option should be secondary, so maybe reposition this below the NPM option?

File: src/routes/docs/quick-starts/web/+page.markdoc [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+// appwrite.ts
+
+import { Client, Databases, Account } from "appwrite";
+// Import type models for Appwrite
+import { type Models } from 'appwrite';
+
+const client: Client = new Client();
+
+client
+    .setEndpoint(<>>)
```
Comment (divanov11) [inline src/routes/docs/quick-starts/web/+page.markdoc:76]: Looks like we have a typo in "setEndpoint"

**Review by divanov11 — COMMENTED** [review]
Added some suggesstions

**Pull request merged**